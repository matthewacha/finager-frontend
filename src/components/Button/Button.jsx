import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';


export default class Button extends PureComponent {
    render(){
        const {
            customButtonStyle,hasText,
            hasIcon, iconImage,iconAlt,iconStyle,
            text,textStyle,disable, onClick} = this.props;
            const isText = !hasText ? hasText : true;
            const isIcon = !hasIcon ? hasIcon : true;
        return(
            <div disable={disable} role="button" onClick={onClick} className={customButtonStyle}>
              {isIcon &&
                <img
                src={iconImage}
                alt={iconAlt}
                className={iconStyle}
                />
                }
              {isText && <span  className={textStyle}>
                {text}
              </span>
              }
            </div>
        );
    }
};

Button.propTypes = {
    customButtonStyle: PropTypes.string.isRequired,
    hasIcon: PropTypes.bool, 
    iconImage: PropTypes.string.isRequired,
    iconAlt: PropTypes.string.isRequired,
    iconStyle: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    textStyle: PropTypes.string.isRequired,
    disable: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    hasText: PropTypes.bool,
}

Button.defaultProps = {
    hasIcon: false,
    disable: "false",
    customButtonStyle: "",
    iconImage: "",
    iconAlt: "",
    iconStyle: "",
    text: "",
    textStyle: "",
    onClick: () => {},
    hasText: false,
}