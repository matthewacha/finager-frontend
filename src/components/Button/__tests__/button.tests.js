import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button.jsx';

describe('Test Button works', () => {
    const props = {
        hasIcon: true,
        iconAlt: "google-icon",
        iconStyle: "google-icon",
        iconImage: "googleIcon",
        customButtonStyle: "auth__panel--social",
        disable: false,
        text: "Login with Google",
        textStyle: "auth-text",
        onClick: jest.fn(),
        hasText: true,
    };

    it('It renders as expected', () => {
        const wrapper = shallow(<Button props={props} />)
        expect(wrapper).toMatchSnapshot();
    })
})