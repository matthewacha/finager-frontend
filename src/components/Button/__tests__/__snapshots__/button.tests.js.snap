// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button works It renders as expected 1`] = `
<div
  className=""
  disable="false"
  onClick={[Function]}
  role="button"
/>
`;
